deposit/migrations/0013_populate_licenses.py

```diff
@@ -0,0 +1,110 @@
+# Generated by Django 2.2.1 on 2019-05-15 14:53
```

Comment:
@wetneb
I'm not sure, if we should keep this migration. For the migration on the production server this is rather nice, since it automatically predefines licenses and connects them with the current repos.

For tests this is not so good, because it adds initial data (that is probably not interfering, cause the licenses are just in the DB and repository entries are not changed since they do not exist).

I think best would be, to remove this migration from PR and manually execute the code via django shell after the migration.

Are you fine with that?

Reply:
I am not sure I fully understand the constraints you have, but my gut feeling is that it is totally fine to add initial data in migrations. What would be the downside of that?

I think we should try to make the deposit interfaces work out of the box as much as possible, and migrations are good at doing that. What I would propose is to add these licenses as migrations in the sub-apps for each protocol (`deposit.zenodo` and `deposit.osf`). This way, the licenses are only created if the protocols are enabled in the settings (as apps).

Reply:
For tests you like to have an empty DB to start with. It's about not have something that could interfere. On the other hand, if you expect the DB to work with the data, tests should work if the data is present.

Technically we could bound the initial licenses to named protocols, although they are meant to be per repo---but I bet that the migration is performed before disabling the protocol in the source.

Reply:
I see your point about tests, but on the other hand I would argue that there is no point testing things in an artificial state where there are no licenses in the database. This model is introduced to replace an hard-coded list, so in all desirable testing scenarios the corresponding table will not be empty. 

Not loading the data in a migration will make our life harder in repository tests (we need to load the licenses explicitly in addition to the other objects) and when deploying an instance (we need to remember to run an extra command to populate this model).

Adding a command to load migrations manually also has the downside of not being extensible: if I want to write a new deposit interface, I will need to modify that command to add the default licenses for that repository. If we use migrations, I just need to add my licenses to a migration in my app.

Reply:
Well, yes, that's unit test and integration test and some different philosophy here on how to do them. That's fine :-)

Let's keep the migration then.